core/commands/store/upload/upload.go

```diff
@@ -0,0 +1,459 @@
+	if err != nil {
+		f.Error(err)
+		return
+	}
+	payerPubKey := payerPrivKey.GetPublic()
+	payinRequest, err := escrow.NewPayinRequest(response, payerPubKey, payerPrivKey)
+	if err != nil {
+		f.Error(err)
+		return
+	}
+	payinRes, err := escrow.PayInToEscrow(f.Context, f.Config, payinRequest)
+	if err != nil {
+		f.Error(fmt.Errorf("failed to pay in to escrow: [%v]", err))
+		return
+	}
+	doGuard(f, payinRes, payerPrivKey)
+}
+
+func doGuard(f *ds.Session, res *escrowpb.SignedPayinResult, payerPriKey ic.PrivKey) {
+	f.Guard()
+	md, err := f.GetMetadata()
+	if err != nil {
+		f.Error(err)
+		return
+	}
+	cts := make([]*guardpb.Contract, 0)
+	for _, h := range md.ShardHashes {
+		shard, err := ds.GetShard(f.PeerId, f.Id, h, &ds.ShardInitParams{
+			Context:   f.Context,
+			Datastore: f.Datastore,
+		})
+		if err != nil {
+			f.Error(err)
+			return
+		}
+		contracts, err := shard.SignedCongtracts()
+		if err != nil {
+			f.Error(err)
+			return
+		}
+		cts = append(cts, contracts.GuardContract)
+	}
+	fsStatus, err := helper.PrepAndUploadFileMeta(f.Context, cts, res, payerPriKey, f.Config, f.PeerId, md.FileHash)
+	if err != nil {
+		f.Error(fmt.Errorf("failed to send file meta to guard: [%v]", err))
+		return
+	}
+
+	qs, err := helper.PrepFileChallengeQuestions(f.Context, f.N, f.Api, fsStatus, md.FileHash, f.PeerId, f.Id)
+	if err != nil {
+		f.Error(err)
+		return
+	}
+
+	fcid, err := cidlib.Parse(md.FileHash)
+	if err != nil {
+		f.Error(err)
+		return
+	}
+	err = helper.SendChallengeQuestions(f.Context, f.Config, fcid, qs)
+	if err != nil {
+		f.Error(fmt.Errorf("failed to send challenge questions to guard: [%v]", err))
+		return
+	}
+	doWaitUpload(f, payerPriKey)
+}
+
+func doWaitUpload(f *ds.Session, payerPriKey ic.PrivKey) {
+	f.WaitUpload()
+	md, err := f.GetMetadata()
+	if err != nil {
+		f.Error(err)
+		return
+	}
+	err = backoff.Retry(func() error {
+		select {
+		case <-f.Context.Done():
+			return nil
+		default:
+		}
+		err := grpc.GuardClient(f.Config.Services.GuardDomain).WithContext(f.Context,
+			func(ctx context.Context, client guardpb.GuardServiceClient) error {
+				req := &guardpb.CheckFileStoreMetaRequest{
+					FileHash:     md.FileHash,
+					RenterPid:    md.RenterId,
+					RequesterPid: f.N.Identity.Pretty(),
+					RequestTime:  time.Now().UTC(),
+				}
+				sign, err := crypto.Sign(payerPriKey, req)
+				if err != nil {
+					return err
+				}
+				req.Signature = sign
+				meta, err := client.CheckFileStoreMeta(ctx, req)
+				if err != nil {
+					return err
+				}
+				fmt.Println("meta.State", meta.State)
+				if meta.State == guardpb.FileStoreStatus_RUNNING {
+					return nil
+				}
+				return errors.New("uploading")
+			})
+		return err
+	}, waitUploadBo)
+	if err != nil {
+		f.Error(err)
+		return
+	}
+	f.Complete()
+}
+
+type UploadRes struct {
+	ID string
+}
+
+func getContractSizeFromCid(ctx context.Context, hash cidlib.Cid, api coreiface.CoreAPI) (uint64, error) {
```

Comment:
contract size? more like node size ... and this should be in some helper file maybe